src/states/game/game.gd

```diff
@@ -369,10 +379,31 @@ func _on_card_dropped(card: DualCard) -> void:
 			else:
 				modified_grid_pos.y -= 3
 			Global.card_current_moves[curr_round].append([script_node.data, modified_grid_pos])
+		var this_unit := get_unit(grid_pos)
+		if this_unit != null and this_unit.card_data.name == "Battering Ram":
+			apply_battering_ram_buff(this_unit)
 		hand.set_all_draggable(true)
 	end_round_button.disabled = false
 
 
+func apply_battering_ram_buff(unit: Unit) -> void:
+	for drop_point in drop_points.get_children():
+		if drop_point.grid_position == unit.grid_position + Vector2i.UP or drop_point.grid_position == unit.grid_position + Vector2i.DOWN:
+			if drop_point.extra_stats.has("attack_power"):
+				drop_point.extra_stats["attack_power"] += int(unit.card_data.extra_data[0])
+			else:
+				drop_point.extra_stats["attack_power"] = int(unit.card_data.extra_data[0])
+
+
+func remove_battering_ram_buff(unit: Unit) -> void:
+	for drop_point in drop_points.get_children():
+		if drop_point.grid_position == unit.grid_position + Vector2i.UP or drop_point.grid_position == unit.grid_position + Vector2i.DOWN:
+			if drop_point.extra_stats.has("attack_power"):
+				drop_point.extra_stats["attack_power"] -= int(unit.card_data.extra_data[0])
+			else:
+				drop_point.extra_stats["attack_power"] = 0
```

Comment:
This is totally valid. Worth noting, we can also just omit these lines since it won't do anything